KubeArmor/core/kubeUpdate.go

```diff
@@ -594,391 +598,411 @@ func (dm *KubeArmorDaemon) HandleUnknownNamespaceNsMap(container *tp.Container)
-						continue
-					}
+	// == Policy == //
 
-					if k == "controller-revision-hash" {
-						continue
-					}
-					pod.Labels[k] = v
-				}
-
-				pod.Containers = map[string]string{}
-				pod.ContainerImages = map[string]string{}
-				for _, container := range event.Object.Status.ContainerStatuses {
-					if len(container.ContainerID) > 0 {
-						cid := strings.Split(container.ContainerID, "://")
-						if len(cid) == 2 { // always true because k8s spec defines format as '<type>://<container_id>'
-							containerID := cid[1]
-							pod.Containers[containerID] = container.Name
-							pod.ContainerImages[containerID] = container.Image + kl.GetSHA256ofImage(container.ImageID)
-						}
-					}
-				}
+	if _, ok := pod.Annotations["kubearmor-policy"]; !ok {
+		pod.Annotations["kubearmor-policy"] = "enabled"
+	}
 
-				// == Policy == //
+	if pod.Annotations["kubearmor-policy"] != "enabled" && pod.Annotations["kubearmor-policy"] != "disabled" && pod.Annotations["kubearmor-policy"] != "audited" {
+		pod.Annotations["kubearmor-policy"] = "enabled"
+	}
 
-				if _, ok := pod.Annotations["kubearmor-policy"]; !ok {
-					pod.Annotations["kubearmor-policy"] = "enabled"
-				}
+	// == LSM == //
 
-				if pod.Annotations["kubearmor-policy"] != "enabled" && pod.Annotations["kubearmor-policy"] != "disabled" && pod.Annotations["kubearmor-policy"] != "audited" {
-					pod.Annotations["kubearmor-policy"] = "enabled"
-				}
+	if dm.RuntimeEnforcer == nil {
+		// exception: no LSM
+		if pod.Annotations["kubearmor-policy"] == "enabled" {
+			pod.Annotations["kubearmor-policy"] = "audited"
+		}
+	} else if dm.RuntimeEnforcer != nil && dm.RuntimeEnforcer.EnforcerType == "SELinux" {
+		// exception: no SELinux support for containers
+		if pod.Annotations["kubearmor-policy"] == "enabled" {
+			pod.Annotations["kubearmor-policy"] = "audited"
+		}
+	}
 
-				// == LSM == //
+	// == Exception == //
 
-				if dm.RuntimeEnforcer == nil {
-					// exception: no LSM
-					if pod.Annotations["kubearmor-policy"] == "enabled" {
-						pod.Annotations["kubearmor-policy"] = "audited"
-					}
-				} else if dm.RuntimeEnforcer != nil && dm.RuntimeEnforcer.EnforcerType == "SELinux" {
-					// exception: no SELinux support for containers
-					if pod.Annotations["kubearmor-policy"] == "enabled" {
-						pod.Annotations["kubearmor-policy"] = "audited"
-					}
-				}
+	// exception: kubernetes app
+	if pod.Metadata["namespaceName"] == "kube-system" {
+		pod.Annotations["kubearmor-policy"] = "audited"
+	}
 
-				// == Exception == //
+	// exception: cilium-operator
+	if _, ok := pod.Labels["io.cilium/app"]; ok {
+		pod.Annotations["kubearmor-policy"] = "audited"
+	}
 
-				// exception: kubernetes app
-				if pod.Metadata["namespaceName"] == "kube-system" {
-					pod.Annotations["kubearmor-policy"] = "audited"
-				}
+	// exception: kubearmor
+	// if _, ok := pod.Labels["kubearmor-app"]; ok {
+	// 	pod.Annotations["kubearmor-policy"] = "audited"
+	// }
```

Comment:
should this be commented?

Reply:
it was done as part of https://github.com/kubearmor/KubeArmor/pull/1831. do you think we can remove all of these exceptions here if it's being handled as part of ignoreNamespace?